count/include/count_min.hpp

```diff
@@ -0,0 +1,189 @@
+  * Only update and estimate methods are added for uint64_t and string types.
+  */
+  count_min_sketch(uint16_t num_hashes, uint32_t num_buckets, uint64_t seed = DEFAULT_SEED) ;
+
+  /**
+  * @return configured num_hashes of this sketch
+  */
+  uint16_t get_num_hashes() const;
+
+  /**
+  * @return configured num_buckets of this sketch
+  */
+  uint32_t get_num_buckets() const;
+
+  /**
+  * @return configured seed of this sketch
+  */
+  uint64_t get_seed()  const;
+
+  /**
+   * @return epsilon : double
+   * The maximum permissible error for any frequency estimate query.
+   * epsilon = ceil(e / num_buckets)
+   */
+   double get_relative_error() ;
+
+  /**
+  * @return total_weight : typename W
+  * The total weight currently inserted into the stream.
+  */
+  W get_total_weight() const;
+
+  /*
+ * @param relative_error : double -- the desired accuracy within which estimates should lie.
+ * For example, when relative_error = 0.05, then the returned frequency estimates satisfy the
+ * `relative_error` guarantee that never overestimates the weights but may underestimate the weights
+ * by 5% of the total weight in the sketch.
+ * @return number_of_buckets : the number of hash buckets at every level of the
+ * sketch required in order to obtain the specified relative error.
+ * [1] - Section 3 ``Data Structure'', page 6.
+ */
+  static uint32_t suggest_num_buckets(double relative_error) ;
+
+  /*
+   * @param confidence : double -- the desired confidence with which estimates should be correct.
+   * For example, with 95% confidence, frequency estimates satisfy the `relative_error` guarantee.
+   * @return number_of_hashes : the number of hash functions that are required in
+   * order to achieve the specified confidence of the sketch.
+   * confidence = 1 - delta, with delta denoting the sketch failure probability in the literature.
+   * [1] - Section 3 ``Data Structure'', page 6.
+  */
+  static uint16_t suggest_num_hashes(double confidence) ;
+
+  /**
+   * Specific get_estimate function for uint64_t type
+   * see generic get_estimate function
+   * @param item : uint64_t type.
+   * @return an estimate of the item's frequency.
+   */
+  W get_estimate(uint64_t item) ;
+
+   /**
+   * Specific get_estimate function for std::string type
+   * see generic get_estimate function
+   * @param item : std::string type
+   * @return an estimate of the item's frequency.
+   */
+  W get_estimate(const std::string& item) ;
+
+  /**
+  * This is the generic estimate query function for any of the given datatypes.
+  * Query the sketch for the estimate of a given item.
+  * @param item : pointer to the data item to be query from the sketch.
+  * @param size : size_t
+  * @return the estimated frequency of the item denoted f_est satisfying
+  * f_true - relative_error*total_weight <= f_est <= f_true
+  */
+   W get_estimate(const void* item, size_t size) ;
```

Comment:
this method should be const, and get_*_bound() below too

Reply:
Resolved.  Had to change `get_hashes(const void* item, size_t size)` to 
`get_hashes(const void* item, size_t size) const;` in order for all of these changes to const to compile correctly.